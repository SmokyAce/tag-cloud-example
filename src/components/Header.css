
.header {
    margin: 50px 50px 0;
}

.header h1 {
    font-size: 3em;
    margin: 0;
}
