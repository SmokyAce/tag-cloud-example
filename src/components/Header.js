import React from 'react';
import PropTypes from 'prop-types';
import './Header.css';

const Header = ({ children, ...props }) => {
  return (
    <div className='header' {...props}>
      <h1>{children}</h1>
    </div>
  );
};

Header.propTypes = {
  children: PropTypes.any
};

export default Header;
