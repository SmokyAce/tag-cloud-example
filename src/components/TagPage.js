import React from 'react';
import PropTypes from 'prop-types';
import Header from './Header';
import TagProperties from './TagProperties';

const TagPage = ({ match: { params } }) => {
  return (
    <React.Fragment>
      <Header>Tag</Header>
      <TagProperties tagId={params.tagId} />
    </React.Fragment>
  );
};

TagPage.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({ tagId: PropTypes.string.isRequired }).isRequired
  }).isRequired
};

export default TagPage;
