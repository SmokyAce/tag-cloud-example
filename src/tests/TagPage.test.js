/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';
import TagPage from '../components/TagPage';
import Header from '../components/Header';

const mockProps = {
  match: {
    params: {
      tagId: '1751295897__Odessa'
    }
  }
};

describe('<TagPage />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<TagPage {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  const testRenderer = TestRenderer.create(<TagPage {...mockProps} />);

  it('renders correctly', () => {
    const snapshot = testRenderer.toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  const testInstance = testRenderer.root;
  expect(testInstance.findByType(Header).props.children).toBe('Tag');
});
