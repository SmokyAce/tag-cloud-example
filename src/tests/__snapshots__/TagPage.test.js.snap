// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TagPage /> renders correctly 1`] = `
Array [
  <div
    className="header"
  >
    <h1>
      Tag
    </h1>
  </div>,
  <div
    id="loading"
  />,
]
`;
